src/ui/controllers/searchbariconcontroller.js

```diff
@@ -16,27 +35,25 @@ export default class SearchBarIconController {
     this.searchBarContainer = config.searchBarContainer;
     this.onMouseUp = this.oneTimeMouseUpListener.bind(this);
 
-    const IconState = {
-      LOADING: new LoadingIconState(this),
-      SEARCH: new SearchIconState(this, config.useCustomIcon),
-      DEFAULT: new DefaultIconState(this, config.useCustomIcon)
-    };
+    const loadingState = new LoadingIconState(this, IconState.LOADING_ICON);
+    const searchState = new SearchIconState(this, IconState.SEARCH_ICON, config.useCustomIcon);
+    const defaultState = new DefaultIconState(this, IconState.DEFAULT_ICON, config.useCustomIcon);
 
-    const transitions = {
-      [IconState.LOADING.id]: {
-        SEARCH_COMPLETE: IconState.DEFAULT
+    const possibleTransitions = {
+      [IconState.LOADING_ICON]: {
+        SEARCH_COMPLETE: defaultState
       },
-      [IconState.DEFAULT.id]: {
-        SUBMIT: IconState.LOADING,
-        FOCUS: IconState.SEARCH
+      [IconState.DEFAULT_ICON]: {
+        SUBMIT: loadingState,
```

Comment:
nit: can we also use the enum for these values as well? (e.g. we would have IconState.LOADING_ICON rather than loadingState) I believe we would need to pass a map to the state machine which maps the enums to the state instances